storage/namespace.go

```diff
@@ -547,12 +594,14 @@ func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
 		if shard == nil {
 			continue
 		}
-		state := shard.FlushState(blockStart)
-		if state.Status == fileOpNotStarted {
-			return true
-		}
-		if state.Status == fileOpFailed && state.NumFailures < maxRetries {
-			return true
+		for _, blockStart := range blockStarts {
```

Comment:
So.. yeah right now it will actually skip flushing the shards that already have flushed - but only  because the Prepare method on the persist manager returns a `nil` persist method if the fileset already exists for a given namespace/shard/time. 

We can make it slightly faster to avoid doing the disk check by calling `shard.FlushState(...)` before `shard.Flush(...)` in the `namespace.Flush(...)` call as suggested.  Fine with doing that.